node/src/chain.rs

```diff
@@ -203,16 +239,44 @@ impl ChainSrv {
                 }
             };
 
+            // Initialize last_finalized block
+            if mrb.header().iteration == 1 || mrb.header().height == 0 {
+                last_finalized = mrb.clone();
+            } else {
+                // scan
+                let mut h = mrb.header().height;
+                loop {
+                    h -= 1;
+                    if let Ok(Some(blk)) = t.fetch_block_by_height(h) {
+                        if blk.header().iteration == 1
+                            || blk.header().height == 0
+                        {
+                            last_finalized = blk;
+                            break;
+                        };
+                    } else {
+                        break;
+                    }
+                }
+            }
+
             Ok(())
         });
 
         tracing::info!(
             event = "Ledger block loaded",
             height = mrb.header().height,
+            finalized_height = last_finalized.header().height,
             hash = hex::encode(mrb.header().hash),
             state_root = hex::encode(mrb.header().state_hash)
         );
 
-        Ok(mrb)
+        Ok((mrb, last_finalized))
+    }
+
+    fn reset_timeout() -> Instant {
```

Comment:
I personally find `reset_timeout` misleading, especially when using as `Self::reset_timeout` because there is nothing to reset.

What about `next_timeout`?

Reply:
Done